addons/jest/src/components/Message.tsx

```diff
@@ -0,0 +1,208 @@
+/* tslint:disable:object-literal-sort-keys */
+
+import React from 'react';
+import { styled } from '@storybook/theming';
+import colors from '../colors';
+
+const patterns = [/^\x08+/, /^\x1b\[[012]?K/, /^\x1b\[?[\d;]{0,3}/];
+
+const Pre = styled.pre({
+  margin: 0,
+});
+
+const Positive = styled.strong({
+  color: colors.success,
+  fontWeight: 500,
+});
+const Negative = styled.strong({
+  color: colors.error,
+  fontWeight: 500,
+});
+
+interface StackTraceProps {
+  trace: MsgElement[];
+  className?: string;
+}
+
+const StackTrace = styled(({ trace, className }: StackTraceProps) => (
+  <details className={className}>
+    <summary>Callstack</summary>
+    {trace
+      .join('')
+      .trim()
+      .split(/\n/)
+      .map((traceLine, traceLineIndex) => (
+        <div key={traceLineIndex}>{traceLine.trim()}</div>
+      ))}
+  </details>
+))({
+  background: 'silver',
+  padding: 10,
+  overflow: 'auto',
+});
+
+const Main = styled(({ msg, className }) => <section className={className}>{msg}</section>)({
+  padding: 10,
+  borderBottom: '1px solid silver',
+});
+
+interface SubProps {
+  msg: MsgElement[];
+  className?: string;
+}
+
+const Sub = styled(({ msg, className }: SubProps) => (
+  <section className={className}>
+    {msg
+      .filter(item => typeof item !== 'string' || item.trim() !== '')
+      .map((item, index, list) => {
+        if (typeof item === 'string') {
```

Comment:
seems odd you filter for anything that isn't a string, then do something for only strings. seems like this code could be clearer

Reply:
It filters for not empty strings, then removes line breaks for those strings